contrib_bots/lib/github_issues.py

```diff
@@ -0,0 +1,126 @@
+import requests
+import json
+import urllib2
+import ConfigParser, os
+import github
+
+class IssueHandler(object):
+    '''
+    This plugin facilitates sending issues to github, when
+    an item is prefixed with '@issue' or '@bug'
+
+    It will also write items to the issues stream, as well
+    as reporting it to github
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow users to flag messages
+            as being issues with Zulip by using te prefix '@issue'
+
+            Before running this, make sure to create a stream
+            called "issues" that your API user can send to.
+
+            Also, make sure that the credentials of the github bot have
+            been typed in correctly, that there is a personal access token
+            with access to public repositories ONLY,
+            and that the repository name is entered correctly.
+
+            Check ~.zuliprc-local, and make sure there are
+            github_repo   (The name of the repo to post to)
+            github_repo_owner   (The owner of the repo to post to)
+            github_username   (The username of the github bot)
+            github_token    (The personal access token for the github bot)
+            '''
+
+    def triage_message(self, message):
+        # return True if we want to (possibly) respond to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'issue':
+            return False
+        is_issue = (original_content.startswith('@issue'))
+
+        return is_issue
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        original_sender = message['sender_email']
+
+        # gets token from dev-secrets
+        # CONFIGURE TOKEN ON LOCAL MACHINE FOR NOW AT .zuliprc-local
+        config = ConfigParser.ConfigParser()
+        config.read([os.path.expanduser('~/.zuliprc-local')])
+
+        REPO_NAME = config.get('api', 'github_repo')
+        REPO_OWNER = config.get('api', 'github_repo_owner')
+
+        new_content = original_content.replace('@issue',
+                                          'by {}:'.format(original_sender,))
+        # gets the repo url
+        url = 'https://api.github.com/repos/{}/{}/issues'.format(REPO_OWNER, REPO_NAME)
+
+        # signs into github using the provided username and password
+        session = github.auth()
+
+        # Gets rid of the @issue in the issue title
+        issue_title = message['content'].replace('@issue', '').strip()
+        issue_content = '' # Initialises the variable that the overflow goes into
```

Comment:
https://www.python.org/dev/peps/pep-0008/#inline-comments

Please take a look at the instruction. There is a suggestion to have at least 2 space between code and `#`. Could you do this?